starport/services/network/campaign.go

```diff
@@ -100,29 +63,61 @@ func (n Network) CampaignUpdateTotalShares(campaignID uint64, totalShares campai
 	return nil
 }
 
-// CampaignUpdateTotalSupply updates the campaign total supply
-func (n Network) CampaignUpdateTotalSupply(campaignID uint64, totalSupply sdk.Coins) error {
-	n.ev.Send(events.New(events.StatusOngoing, fmt.Sprintf(
-		"Updating the campaign %d total supply to %s",
-		campaignID,
-		totalSupply.String(),
-	)))
+// CampaignEdit updates the campaign name or metadata
+func (n Network) CampaignEdit(
```

Comment:
CampaignEdit is very similar to [this](https://github.com/tendermint/spn/blob/2ef24d4/proto/campaign/tx.proto#L14) func in naming. 

But here it doesn't just stand for that func and also stands for supply related funcs. 

I believe we should use a different name something like `UpdateCampaign` to avoid any confision. 

Also, can we please use option syntax, as described [here](https://github.com/tendermint/starport/pull/2099#pullrequestreview-891833685)?

Reply:
The current implementation is similar, but we use the object directly instead of wrapping the `sdk.Msg` into a `prop` object. What is your thought about the `props` struct?

```go
type props struct {
	msg sdk.Msg
}
```

or 

```go
type props struct {
	campaignID  uint64
	name        string
	metadata    []byte
	totalShares campaigntypes.Shares
	totalSupply sdk.Coins
}
```

Reply:
```go
UpdateCampaign(id uint64, prop Prop, props ...Prop)

func WithTotalShares(shares campaigntypes.Shares) Prop {
    return func(c *updateConfig) {
        c.totalShares = shares
   }
}

type updateConfig struct {
        name        string
	metadata    []byte
	totalShares campaigntypes.Shares
	totalSupply sdk.Coins
}
```